src/test/shared/telemetry/telemetryUtils.test.ts

```diff
@@ -0,0 +1,64 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import { Disposable } from 'vscode'
+import { ext } from '../../../shared/extensionGlobals'
+import { TelemetryEvent } from '../../../shared/telemetry/telemetryEvent'
+import { TelemetryService } from '../../../shared/telemetry/telemetryService'
+import { registerCommand } from '../../../shared/telemetry/telemetryUtils'
+
+class MockTelemetryService implements TelemetryService {
+    public persistFilePath: string = ''
+    public lastEvent: TelemetryEvent | undefined
+    private _telemetryEnabled: boolean = false
+
+    public get telemetryEnabled(): boolean {
+        return this._telemetryEnabled
+    }
+    public set telemetryEnabled(value: boolean) {
+        this._telemetryEnabled = value
+    }
+
+    public async start(): Promise<void> { return }
+    public async shutdown(): Promise<void> { return }
+    public record(event: TelemetryEvent ): void { this.lastEvent = event }
+    public clearRecords(): void {}
+    public notifyOptOutOptionMade(): void {}
+}
+
+describe('telemetryUtils', () => {
+    describe('registerCommand', () => {
+        const mockService = new MockTelemetryService()
+        ext.telemetry = mockService
+
+        it('records telemetry', (done) => {
+            registerCommand({
+                register: (_command, callback: (...args: any[]) => Promise<void>, _thisArg) => {
+                    // vscode.commands.registerCommand is not async, but we can't check until the callback is complete
+                    callback().then(() => {
+                        assert.notStrictEqual(mockService.lastEvent, undefined)
```

Comment:
Do you need to fire the command in order to exercise these asserts?

Reply:
yup